File: solutions/go/sieve/1/sieve.go
package sieve

func Sieve(limit int) []int {
    if limit < 2 {
        return []int{}
    }
	primes := []int{2}
    for i := 2; i <= limit; i++ {
        isPrime := true
        for _, p := range primes {
            if i % p == 0 {
                isPrime = false
                break
            }
        }
        if isPrime {
            primes = append(primes, i)
        }
    }
    return primes
}

File: solutions/cpp/sieve/1/sieve.cpp
#include "sieve.h"

namespace sieve {

    const std::vector<int> primes(const int n) {
        std::vector<int> result(0);
        if (n < 2) {
            return result;
        }
        result.push_back(2);
        bool is_prime{};
        for (int cur = 3; cur <= n; ++cur) {
            is_prime = true;
            for (auto prime = result.begin(); prime != result.end(); ++prime) {
                if (cur % *prime == 0) {
                    is_prime = false;
                    break;
                }
            }
            if (is_prime) {
                result.push_back(cur);
            }
        }
        return result;
    }

}  // namespace sieve
